lib/AuthStateManager.ts

```diff
@@ -100,7 +100,8 @@ export class AuthStateManager {
     };
 
     const emitAuthStateChange = (authState) => {
-      if (isSameAuthState(this._authState, authState)) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
The main reason (when working in 5.x branch) was to avoid making any breaking changes. Now that it is in 6.x branch we can think about making some changes. I believe it is possible that `this._authState` could be null in this case so I'll revise `isSameAuthState` to accept a null value